### PR TITLE
fix: bug #1869, DML .setFitIntercept should be set to true

### PR DESCRIPTION
**Pull request by dylanw-oss**
Fix bug #1869, and add colTyple and ModelType check

### CODE REVIEW

**Review by mhamilton723 — DISMISSED** [review]

File: core/src/main/scala/com/microsoft/azure/synapse/ml/causal/DoubleMLEstimator.scala
Comment (mhamilton723) [inline core/src/main/scala/com/microsoft/azure/synapse/ml/causal/DoubleMLEstimator.scala:247]: would this model benefit from exposing all of these as params with these defaults?

Comment (dylanw-oss) [reply core/src/main/scala/com/microsoft/azure/synapse/ml/causal/DoubleMLEstimator.scala:247]: @sarahshy

Comment (sarahshy) [reply core/src/main/scala/com/microsoft/azure/synapse/ml/causal/DoubleMLEstimator.scala:247]: I think we should keep the parameters fixed and hidden since there won't (at least, shouldn't) be a case where they need to be changed.

**Comment by acrolinxatmsft1** [comment]
## Acrolinx Scorecards

**A minimum Acrolinx score of 80 is required.**

Click the scorecard links for each article to review the Acrolinx feedback on grammar, spelling, punctuation, writing style, and terminology.

| Article | Acrolinx<br>score | Word and<br>Phrases Score | Correctness<br>Score | Scorecard | Processed |
|---------|:--------------:|:--------------------:|:------:|:---------:|:---------:|
| [website/docs/documentation/estimators/causal/_causalInferenceDML.md](https://github.com/microsoft/SynapseML/blob/bb83c148fc8138263f2d0d4dc92821063194d764/website%2Fdocs%2Fdocumentation%2Festimators%2Fcausal%2F_causalInferenceDML.md) | 100 | 100 | 100 | [link](https://microsoft-ce-csi.acrolinx.cloud/api/v1/checking/scorecards/3790d3b7-435a-49e7-8765-2c285445652e) | [:white_check_mark:](# "Successfully processed") |

**More information about Acrolinx**

- [Install Acrolinx locally for VSCode](https://review.docs.microsoft.com/help/contribute/contribute-acrolinx-vscode)
- [False positives or issues](https://aka.ms/acrolinxbug)
- [Troubleshooting issues with Acrolinx](https://review.docs.microsoft.com/help/contribute/acrolinx-error-messages)

**Comment by acrolinxatmsft1** [comment]
## Acrolinx Scorecards

**A minimum Acrolinx score of 80 is required.**

Click the scorecard links for each article to review the Acrolinx feedback on grammar, spelling, punctuation, writing style, and terminology.

| Article | Acrolinx<br>score | Word and<br>Phrases Score | Correctness<br>Score | Scorecard | Processed |
|---------|:--------------:|:--------------------:|:------:|:---------:|:---------:|
| [website/docs/documentation/estimators/causal/_causalInferenceDML.md](https://github.com/microsoft/SynapseML/blob/9b955556354462a01a2e2d21ec9e87040f45cdaa/website%2Fdocs%2Fdocumentation%2Festimators%2Fcausal%2F_causalInferenceDML.md) | 100 | 100 | 100 | [link](https://microsoft-ce-csi.acrolinx.cloud/api/v1/checking/scorecards/7afa2228-3e8f-46b6-b260-5842eae61c10) | [:white_check_mark:](# "Successfully processed") |

**More information about Acrolinx**

- [Install Acrolinx locally for VSCode](https://review.docs.microsoft.com/help/contribute/contribute-acrolinx-vscode)
- [False positives or issues](https://aka.ms/acrolinxbug)
- [Troubleshooting issues with Acrolinx](https://review.docs.microsoft.com/help/contribute/acrolinx-error-messages)

**Comment by acrolinxatmsft1** [comment]
## Acrolinx Scorecards

**A minimum Acrolinx score of 80 is required.**

Click the scorecard links for each article to review the Acrolinx feedback on grammar, spelling, punctuation, writing style, and terminology.

| Article | Acrolinx<br>score | Word and<br>Phrases Score | Correctness<br>Score | Scorecard | Processed |
|---------|:--------------:|:--------------------:|:------:|:---------:|:---------:|
| [website/docs/documentation/estimators/causal/_causalInferenceDML.md](https://github.com/microsoft/SynapseML/blob/8ddf99749cd6ff7ea0c518bd48606bce83303e60/website%2Fdocs%2Fdocumentation%2Festimators%2Fcausal%2F_causalInferenceDML.md) | 100 | 100 | 100 | [link](https://microsoft-ce-csi.acrolinx.cloud/api/v1/checking/scorecards/40c261e8-3e08-450e-9f23-72396004367f) | [:white_check_mark:](# "Successfully processed") |

**More information about Acrolinx**

- [Install Acrolinx locally for VSCode](https://review.docs.microsoft.com/help/contribute/contribute-acrolinx-vscode)
- [False positives or issues](https://aka.ms/acrolinxbug)
- [Troubleshooting issues with Acrolinx](https://review.docs.microsoft.com/help/contribute/acrolinx-error-messages)

**Comment by acrolinxatmsft1** [comment]
## Acrolinx Scorecards

**A minimum Acrolinx score of 80 is required.**

Click the scorecard links for each article to review the Acrolinx feedback on grammar, spelling, punctuation, writing style, and terminology.

| Article | Acrolinx<br>score | Word and<br>Phrases Score | Correctness<br>Score | Scorecard | Processed |
|---------|:--------------:|:--------------------:|:------:|:---------:|:---------:|
| [website/docs/documentation/estimators/causal/_causalInferenceDML.md](https://github.com/microsoft/SynapseML/blob/bbb89bf421835955d0dcd94a151c5d9f81b01124/website%2Fdocs%2Fdocumentation%2Festimators%2Fcausal%2F_causalInferenceDML.md) | 100 | 100 | 100 | [link](https://microsoft-ce-csi.acrolinx.cloud/api/v1/checking/scorecards/a5ce8a7c-bc18-42c7-baa1-c1715ec1465a) | [:white_check_mark:](# "Successfully processed") |

**More information about Acrolinx**

- [Install Acrolinx locally for VSCode](https://review.docs.microsoft.com/help/contribute/contribute-acrolinx-vscode)
- [False positives or issues](https://aka.ms/acrolinxbug)
- [Troubleshooting issues with Acrolinx](https://review.docs.microsoft.com/help/contribute/acrolinx-error-messages)

**Review by sarahshy — APPROVED** [review]
Looks good to me, but I defer to the code owners for the technical code review.

**Review by mhamilton723 — APPROVED** [review]

**Pull request merged**
